### PR TITLE
perf: fetching projects page

### PR DESCRIPTION
**Pull request by victordelval**
## Description

Closes: regen-network/regen-registry#1345

* Add projects page loader

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1637--regen-registry.netlify.app/projects

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by mhagel — APPROVED** [review]
Very noticeable load time improvement!

**Review by flagrede — COMMENTED** [review]
Looks good!
I just noticed that queries that end up in error like 404 were still triggered every time we go on the page. Could you check if there's a way to prevent that?
It's mostly metadata calls apparently, a try-catch that returns an empty result instead should do it.

**Comment by victordelval** [comment]
> 

done! just returning null. Awesome!

**Comment by victordelval** [comment]
> Looks good! I just noticed that queries that end up in error like 404 were still triggered every time we go on the page. Could you check if there's a way to prevent that? It's mostly metadata calls apparently, a try-catch that returns an empty result instead should do it.

done! just returning null. Awesome!

File: web-registry/src/pages/Projects/Projects.loader.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const simplePriceQuery = getSimplePriceQuery({});
+
+    // Fetch or Cache
+
+    // Mandatory data
+    await getFromCacheOrFetch({
+      query: projectsQuery,
+      reactQueryClient: queryClient,
+    });
+    // Optionnal data
```
Comment (blushi) [inline web-registry/src/pages/Projects/Projects.loader.ts:40]:
```suggestion
    // Optional data
```

**Review by blushi — APPROVED** [review]
LGTM

**Review by flagrede — APPROVED** [review]
Looks good, thanks!

**Comment by victordelval** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**